### DISCUSSION TITLE
Regex as path

### DISCUSSION THREAD

**Discussion by timreichen** [oakserver/oak#365] [Ideas]
It would be nice to be able to specify a regex as a path. Since a string gets turned into a regex in the `Layer` constructor anyway, i think this would not break any functionality under the hood and would make the framework more expressive. I don't quite see the point in specifying a regex as a string if js has a regex API, but they also could coexist.
```ts
router.get(/(.*)/, (context) => { … });
```
instead of
```ts
router.get("(.*)", (context) => { … });
```

**Comment by kitsonk** [comment]
[path-to-regex](https://github.com/pillarjs/path-to-regexp) is a well established pattern/library for middleware routers like oak.  The problem with plain regexs is what to do with the capturing groups, yes, some people name then, but it is far more cumbersome.  When expressing a path match, it is a lot better/easier to express it as a path-to-regexp than it is a regular expression.  Neither Express nor Koa routers accepts regexs for paths.

In addition, is someone wants to use a regex match the path, they can just use middleware that checks the HTTP verb and/or path. The router is really intended to match structured routes.

Basically I am not very keen on it.

**Reply by timreichen** [reply]
Thanks for the fast reply.
I see that path-to-regex is an established de facto standard, though using it or creating naming groups in a regex is more about taste in my opinion.
I disagree that it is just better to use path-to-regexp but agree that it might be easier for general use cases. But in some cases it would really make sense to be able to use regexs. For example if you already have a regex created and used elsewhere in your code and want to have it as a router path, which is a scenario i ran into. It does not make sense to regex->string->regex.
Since path-to-regex is an abstraction level, I don't see why a lower level API shouln't be supported, since it wouldn't break anything.

**Reply by kitsonk** [reply]
It would "break" things in the sense of how would router props work with a regex?
